## Get back to point in course where start quiz from

In [89]:
import graphlab

Set canvas target

In [90]:
graphlab.canvas.set_target('ipynb')

### Load the data

In [91]:
products = graphlab.SFrame('amazon_baby.gl/')

In [92]:
products

Columns:
	name	str
	review	str
	rating	float

Rows: 183531

Data:
+-------------------------------+-------------------------------+--------+
|              name             |             review            | rating |
+-------------------------------+-------------------------------+--------+
|    Planetwise Flannel Wipes   | These flannel wipes are OK... |  3.0   |
|     Planetwise Wipe Pouch     | it came early and was not ... |  5.0   |
| Annas Dream Full Quilt wit... | Very soft and comfortable ... |  5.0   |
| Stop Pacifier Sucking with... | This is a product well wor... |  5.0   |
| Stop Pacifier Sucking with... | All of my kids have cried ... |  5.0   |
| Stop Pacifier Sucking with... | When the Binky Fairy came ... |  5.0   |
| A Tale of Baby's Days with... | Lovely book, it's bound ti... |  4.0   |
| Baby Tracker&reg; - Daily ... | Perfect for new parents. W... |  5.0   |
| Baby Tracker&reg; - Daily ... | A friend of mine pinned th... |  5.0   |
| Baby Tracker&reg; - Daily ... | This has been an easy way ... |  4.0   |
+-------------------------------+-------------------------------+--------+
[183531 rows x 3 columns]
Note: Only the head of the SFrame is printed.
You can use print_rows(num_rows=m, num_columns=n) to print more rows and columns.

#### Delete 3 as is a neutral rating

In [93]:
products = products[products['rating'] != 3]

#### Create new column for sentiment where true is 4 or 5, else false

In [94]:
products['sentiment'] = products['rating'] >= 4

## Create the word count vector for each word in the review
Set the target column that tells us whether is a good or bad review

In [95]:
products['word_count'] = graphlab.text_analytics.count_words(products['review'])

In [96]:
products.head()

Columns:
	name	str
	review	str
	rating	float
	sentiment	int
	word_count	dict

Rows: 10

Data:
+-------------------------------+-------------------------------+--------+-----------+
|              name             |             review            | rating | sentiment |
+-------------------------------+-------------------------------+--------+-----------+
|     Planetwise Wipe Pouch     | it came early and was not ... |  5.0   |     1     |
| Annas Dream Full Quilt wit... | Very soft and comfortable ... |  5.0   |     1     |
| Stop Pacifier Sucking with... | This is a product well wor... |  5.0   |     1     |
| Stop Pacifier Sucking with... | All of my kids have cried ... |  5.0   |     1     |
| Stop Pacifier Sucking with... | When the Binky Fairy came ... |  5.0   |     1     |
| A Tale of Baby's Days with... | Lovely book, it's bound ti... |  4.0   |     1     |
| Baby Tracker&reg; - Daily ... | Perfect for new parents. W... |  5.0   |     1     |
| Baby Tracker&reg; - Daily ... | A friend of mine pinned th... |  5.0   |     1     |
| Baby Tracker&reg; - Daily ... | This has been an easy way ... |  4.0   |     1     |
| Baby Tracker&reg; - Daily ... | I love this journal and ou... |  4.0   |     1     |
+-------------------------------+-------------------------------+--------+-----------+
+-------------------------------+
|           word_count          |
+-------------------------------+
| {'and': 3, 'love': 1, 'it'... |
| {'and': 2, 'quilt': 1, 'it... |
| {'ingenious': 1, 'and': 3,... |
| {'and': 2, 'parents!!': 1,... |
| {'and': 2, 'this': 2, 'her... |
| {'shop': 1, 'noble': 1, 'i... |
| {'and': 2, 'all': 1, 'righ... |
| {'and': 1, 'help': 1, 'giv... |
| {'journal.': 1, 'nanny': 1... |
| {'all': 1, 'forget': 1, 'j... |
+-------------------------------+
[10 rows x 5 columns]

# 1) Word count

## Create transform to extract count of a word

Use a function and use lambda to pass multipe variables
https://stackoverflow.com/questions/33204763/how-to-pass-multiple-arguments-to-the-apply-function

In [97]:
def counts_words(word_count, word):
    if word in word_count:
        return word_count[word]
    else:
        return 0

In [98]:
products['word_count'].apply(lambda dict: counts_words(dict, 'and'))

dtype: int
Rows: 166752
[3, 2, 3, 2, 2, 0, 2, 1, 0, 5, 1, 5, 1, 3, 1, 1, 5, 2, 3, 1, 4, 2, 2, 1, 2, 0, 2, 1, 3, 2, 0, 6, 1, 0, 0, 3, 3, 4, 2, 2, 1, 7, 0, 2, 5, 0, 0, 0, 1, 1, 4, 0, 0, 2, 0, 4, 0, 2, 2, 1, 1, 1, 2, 0, 1, 3, 1, 0, 4, 2, 2, 1, 1, 1, 0, 3, 4, 0, 0, 0, 0, 0, 0, 3, 2, 2, 2, 1, 0, 1, 2, 3, 0, 2, 2, 4, 1, 6, 1, 1, ... ]

Or you could use the build in dictionary function called .get() which does the same as the above

In [99]:
products['word_count'].apply(lambda dict: dict.get('and', 0))

dtype: int
Rows: 166752
[3, 2, 3, 2, 2, 0, 2, 1, 0, 5, 1, 5, 1, 3, 1, 1, 5, 2, 3, 1, 4, 2, 2, 1, 2, 0, 2, 1, 3, 2, 0, 6, 1, 0, 0, 3, 3, 4, 2, 2, 1, 7, 0, 2, 5, 0, 0, 0, 1, 1, 4, 0, 0, 2, 0, 4, 0, 2, 2, 1, 1, 1, 2, 0, 1, 3, 1, 0, 4, 2, 2, 1, 1, 1, 0, 3, 4, 0, 0, 0, 0, 0, 0, 3, 2, 2, 2, 1, 0, 1, 2, 3, 0, 2, 2, 4, 1, 6, 1, 1, ... ]

### awesome

In [100]:
products['awesome'] = products['word_count'].apply(lambda dict: dict.get('awesome', 0))

In [101]:
products['awesome'].sum()

2002

### great

In [102]:
products['great'] = products['word_count'].apply(lambda dict: dict.get('great', 0))

In [103]:
products['great'].sum()

42420

### fantastic

In [104]:
products['fantastic'] = products['word_count'].apply(lambda dict: dict.get('fantastic', 0))

In [105]:
products['fantastic'].sum()

873

### amazing 

In [106]:
products['amazing'] = products['word_count'].apply(lambda dict: dict.get('amazing', 0))

In [107]:
products['amazing'].sum()

1305

### love

In [108]:
products['love'] = products['word_count'].apply(lambda dict: dict.get('love', 0))

In [109]:
products['love'].sum()

40277

### horrible


In [110]:
products['horrible'] = products['word_count'].apply(lambda dict: dict.get('horrible', 0))

In [111]:
products['horrible'].sum()

659

### bad

In [112]:
products['bad'] = products['word_count'].apply(lambda dict: dict.get('bad', 0))

In [113]:
products['bad'].sum()

3197

### terrible

In [114]:
products['terrible'] = products['word_count'].apply(lambda dict: dict.get('terrible', 0))

In [115]:
products['terrible'].sum()

673

### awful

In [116]:
products['awful'] = products['word_count'].apply(lambda dict: dict.get('awful', 0))

In [117]:
products['awful'].sum()

345

### wow

In [118]:
products['wow'] = products['word_count'].apply(lambda dict: dict.get('wow', 0))

In [119]:
products['wow'].sum()

131

### hate

In [120]:
products['hate'] = products['word_count'].apply(lambda dict: dict.get('hate', 0))

In [121]:
products['hate'].sum()

1057

# Answer 1) 

## Min: wow=144

## Max: great=42420

# 2) Most Positive and Negative Weighting

In [122]:
selected_words = ['awesome', 'great', 'fantastic', 'amazing', 'love', 'horrible', 'bad', 'terrible', 'awful', 'wow', 'hate']

Create training data

In [124]:
train_data, test_data = products.random_split(.8,seed=0)

Now using array as features column (were using word_count coumn only before)

In [126]:
selected_words_model  = graphlab.logistic_classifier.create(train_data, 
                                                     target='sentiment',
                                                     features=selected_words,
                                                     validation_set=train_data)

Logistic regression:

--------------------------------------------------------

Number of examples          : 133448

Number of classes           : 2

Number of feature columns   : 11

Number of unpacked features : 11

Number of coefficients    : 12

Starting Newton Method

--------------------------------------------------------

+-----------+----------+--------------+-------------------+---------------------+

| Iteration | Passes   | Elapsed Time | Training-accuracy | Validation-accuracy |

+-----------+----------+--------------+-------------------+---------------------+

| 1         | 2        | 0.359932     | 0.844299          | 0.844299            |

| 2         | 3        | 0.580601     | 0.844186          | 0.844186            |

| 3         | 4        | 0.848995     | 0.844276          | 0.844276            |

| 4         | 5        | 1.058655     | 0.844269          | 0.844269            |

| 5         | 6        | 1.257658     | 0.844269          | 0.844269            |

| 6         | 7        | 1.475394     | 0.844269          | 0.844269            |

+-----------+----------+--------------+-------------------+---------------------+

SUCCESS: Optimal solution found.

Examine the weights the learned classifier assigned to each of the 11 words 

In [188]:
sorted_coefficients = selected_words_model['coefficients'].sort('value',ascending=False)

In [190]:
sorted_coefficients.print_rows(num_rows=12)

+-------------+-------+-------+------------------+------------------+
|     name    | index | class |      value       |      stderr      |
+-------------+-------+-------+------------------+------------------+
|     love    |  None |   1   |  1.39989834302   | 0.0287147460124  |
| (intercept) |  None |   1   |  1.36728315229   | 0.00861805467824 |
|   awesome   |  None |   1   |  1.05800888878   |  0.110865296265  |
|   amazing   |  None |   1   |  0.892802422508  |  0.127989503231  |
|  fantastic  |  None |   1   |  0.891303090304  |  0.154532343591  |
|    great    |  None |   1   |  0.883937894898  | 0.0217379527921  |
|     wow     |  None |   1   | -0.0541450123333 |  0.275616449416  |
|     bad     |  None |   1   | -0.985827369929  | 0.0433603009142  |
|     hate    |  None |   1   |  -1.40916406276  | 0.0771983993506  |
|    awful    |  None |   1   |  -1.76469955631  |  0.134679803365  |
|   horrible  |  None |   1   |  -1.99651800559  | 0.0973584169028  |
|   terrible  |  Non

In [145]:
sorted_coefficients.head(1)

Columns:
	name	str
	index	str
	class	int
	value	float
	stderr	float

Rows: 1

Data:
+------+-------+-------+---------------+-----------------+
| name | index | class |     value     |      stderr     |
+------+-------+-------+---------------+-----------------+
| love |  None |   1   | 1.39989834302 | 0.0287147460124 |
+------+-------+-------+---------------+-----------------+
[1 rows x 5 columns]

In [144]:
sorted_coefficients.tail(1)

Columns:
	name	str
	index	str
	class	int
	value	float
	stderr	float

Rows: 1

Data:
+----------+-------+-------+----------------+-----------------+
|   name   | index | class |     value      |      stderr     |
+----------+-------+-------+----------------+-----------------+
| terrible |  None |   1   | -2.09049998487 | 0.0967241912229 |
+----------+-------+-------+----------------+-----------------+
[1 rows x 5 columns]

## Most Negative: 'terrible'=-2.090499984872607
## Most Positive: 'love'=1.39989834302
Makes sense since more reviews would give high scores for love

# 3) Comparing Accuracy

In [146]:
selected_words_model.evaluate(test_data)

{'accuracy': 0.8431419649291376,
 'auc': 0.6648096413721418,
 'confusion_matrix': Columns:
 	target_label	int
 	predicted_label	int
 	count	int
 
 Rows: 4
 
 Data:
 +--------------+-----------------+-------+
 | target_label | predicted_label | count |
 +--------------+-----------------+-------+
 |      0       |        0        |  234  |
 |      0       |        1        |  5094 |
 |      1       |        1        | 27846 |
 |      1       |        0        |  130  |
 +--------------+-----------------+-------+
 [4 rows x 3 columns],
 'f1_score': 0.914242563530107,
 'log_loss': 0.4054747110365649,
 'precision': 0.8453551912568306,
 'recall': 0.9953531598513011,
 'roc_curve': Columns:
 	threshold	float
 	fpr	float
 	tpr	float
 	p	int
 	n	int
 
 Rows: 100001
 
 Data:
 +-----------+-----+-----+-------+------+
 | threshold | fpr | tpr |   p   |  n   |
 +-----------+-----+-----+-------+------+
 |    0.0    | 1.0 | 1.0 | 27976 | 5328 |
 |   1e-05   | 1.0 | 1.0 | 27976 | 5328 |
 |   2e-05   | 

## Accuracy of 11 words = 0.8431419649291376

Recreate the model from the lecture on all data

In [148]:
sentiment_model = graphlab.logistic_classifier.create(train_data, 
                                                     target='sentiment',
                                                     features=['word_count'],
                                                     validation_set=train_data)

WARNING: The number of feature dimensions in this problem is very large in comparison with the number of examples. Unless an appropriate regularization value is set, this model may not provide accurate predictions for a validation/test set.

Logistic regression:

--------------------------------------------------------

Number of examples          : 133448

Number of classes           : 2

Number of feature columns   : 1

Number of unpacked features : 219217

Number of coefficients    : 219218

Starting L-BFGS

--------------------------------------------------------

+-----------+----------+-----------+--------------+-------------------+---------------------+

| Iteration | Passes   | Step size | Elapsed Time | Training-accuracy | Validation-accuracy |

+-----------+----------+-----------+--------------+-------------------+---------------------+

| 1         | 5        | 0.000002  | 2.228286     | 0.841481          | 0.841481            |

| 2         | 9        | 3.000000  | 4.661635     | 0.947425          | 0.947425            |

| 3         | 10       | 3.000000  | 5.769835     | 0.923768          | 0.923768            |

| 4         | 11       | 3.000000  | 6.942404     | 0.971779          | 0.971779            |

| 5         | 12       | 3.000000  | 8.040379     | 0.975511          | 0.975511            |

| 6         | 13       | 3.000000  | 9.017792     | 0.899991          | 0.899991            |

| 7         | 15       | 1.000000  | 10.469377    | 0.984548          | 0.984548            |

| 8         | 16       | 1.000000  | 11.510268    | 0.985118          | 0.985118            |

| 9         | 17       | 1.000000  | 12.473259    | 0.987066          | 0.987066            |

| 10        | 18       | 1.000000  | 13.513086    | 0.988715          | 0.988715            |

+-----------+----------+-----------+--------------+-------------------+---------------------+

TERMINATED: Iteration limit reached.

This model may not be optimal. To improve it, consider increasing `max_iterations`.

In [149]:
sentiment_model.evaluate(test_data)

{'accuracy': 0.916256305548883,
 'auc': 0.9446492867438502,
 'confusion_matrix': Columns:
 	target_label	int
 	predicted_label	int
 	count	int
 
 Rows: 4
 
 Data:
 +--------------+-----------------+-------+
 | target_label | predicted_label | count |
 +--------------+-----------------+-------+
 |      0       |        1        |  1328 |
 |      0       |        0        |  4000 |
 |      1       |        1        | 26515 |
 |      1       |        0        |  1461 |
 +--------------+-----------------+-------+
 [4 rows x 3 columns],
 'f1_score': 0.9500349343413533,
 'log_loss': 0.26106698432422465,
 'precision': 0.9523039902309378,
 'recall': 0.9477766657134686,
 'roc_curve': Columns:
 	threshold	float
 	fpr	float
 	tpr	float
 	p	int
 	n	int
 
 Rows: 100001
 
 Data:
 +-----------+----------------+----------------+-------+------+
 | threshold |      fpr       |      tpr       |   p   |  n   |
 +-----------+----------------+----------------+-------+------+
 |    0.0    |      1.0       | 

## Accuracy of all words = 0.916256305548883

This is just like in the lecture when showing the evaluation graph

Total of all sums = 92,939

Majority Classifier is 'great' = 42420

So even just considering the 11 words this is less than 50% so 90% found above is good.
No class inbalance
Random is 1/11 = 9%

### Quiz question 7) Which of the following ranges contains the accuracy of the majority class classifier, which simply predicts the majority class on the test_data?

Notice it says *test_data*

So as per this discussion: https://www.coursera.org/learn/ml-foundations/module/LJsGo/discussions/kL-vDexTEeWTdBIkpCpI1Q To find the major class classifier from the test data, based in the column 'sentiment'. Just sum the entire column (to find the number of items classified as 1) and divide by the lenght of the SFrame

In [201]:
test_data['sentiment'].sum()

27976

In [197]:
test_data.num_rows

<bound method SFrame.num_rows of Columns:
	name	str
	review	str
	rating	float
	sentiment	int
	word_count	dict
	awesome	int
	great	int
	fantastic	int
	amazing	int
	love	int
	horrible	int
	bad	int
	terrible	int
	awful	int
	wow	int
	hate	int

Rows: 33304

Data:
+-------------------------------+-------------------------------+--------+-----------+
|              name             |             review            | rating | sentiment |
+-------------------------------+-------------------------------+--------+-----------+
| Nature's Lullabies First Y... | Space for monthly photos, ... |  5.0   |     1     |
| Nature's Lullabies Second ... | This was the only calender... |  5.0   |     1     |
|  Lamaze Peekaboo, I Love You  | I like how the book has a ... |  5.0   |     1     |
|  Lamaze Peekaboo, I Love You  | What a great book for babi... |  5.0   |     1     |
| SoftPlay Twinkle Twinkle E... |                               |  5.0   |     1     |
| SoftPlay Peek-A-Boo Where'... | We wanted t

In [205]:
27976 / float(33304)

0.8400192169108815

# Answers)

Accuracy of 11 words = 0.8431419649291376

Accuracy of all words = 0.916256305548883

Accurcy of majority class = 0.8400192169108815

# 4) Interpreting the difference in performance between the models

select all reviews where the product name is ‘Baby Trend Diaper Champ’.

In [157]:
diaper_champ_reviews = products[products['name'] == 'Baby Trend Diaper Champ']

In [158]:
diaper_champ_reviews.head()

Columns:
	name	str
	review	str
	rating	float
	sentiment	int
	word_count	dict
	awesome	int
	great	int
	fantastic	int
	amazing	int
	love	int
	horrible	int
	bad	int
	terrible	int
	awful	int
	wow	int
	hate	int

Rows: 10

Data:
+-------------------------+-------------------------------+--------+-----------+
|           name          |             review            | rating | sentiment |
+-------------------------+-------------------------------+--------+-----------+
| Baby Trend Diaper Champ | Ok - newsflash.  Diapers a... |  4.0   |     1     |
| Baby Trend Diaper Champ | My husband and I selected ... |  1.0   |     0     |
| Baby Trend Diaper Champ | Excellent diaper disposal ... |  5.0   |     1     |
| Baby Trend Diaper Champ | We love our diaper champ. ... |  5.0   |     1     |
| Baby Trend Diaper Champ | Two girlfriends and two fa... |  5.0   |     1     |
| Baby Trend Diaper Champ | I waited to review this un... |  4.0   |     1     |
| Baby Trend Diaper Champ | I have had a diaper genie ... |  1.0   |     0     |
| Baby Trend Diaper Champ | I originally put this item... |  5.0   |     1     |
| Baby Trend Diaper Champ | I am so glad I got the Dia... |  5.0   |     1     |
| Baby Trend Diaper Champ | We had 2 diaper Genie's bo... |  4.0   |     1     |
+-------------------------+-------------------------------+--------+-----------+
+-------------------------------+---------+-------+-----------+---------+------+
|           word_count          | awesome | great | fantastic | amazing | love |
+-------------------------------+---------+-------+-----------+---------+------+
| {'son': 1, 'just': 2, 'les... |    0    |   0   |     0     |    0    |  0   |
| {'material)': 1, 'bags,': ... |    0    |   0   |     0     |    0    |  0   |
| {'control': 1, 'am': 1, 'i... |    0    |   0   |     0     |    0    |  0   |
| {'and': 3, 'over.': 1, 'al... |    0    |   0   |     0     |    0    |  1   |
| {'just': 1, '-': 3, 'both'... |    0    |   0   |     0     |    0    |  0   |
| {'lysol': 1, 'all': 1, 'mo... |    0    |   0   |     0     |    0    |  0   |
| {'all': 1, 'bags.': 1, 'ju... |    0    |   0   |     0     |    0    |  0   |
| {'lysol': 1, 'all': 2, 'ba... |    0    |   0   |     0     |    0    |  0   |
| {'and': 2, 'all': 1, 'just... |    0    |   0   |     0     |    0    |  0   |
| {'hand.': 1, 'both': 1, '(... |    0    |   0   |     0     |    0    |  2   |
+-------------------------------+---------+-------+-----------+---------+------+
+----------+-----+----------+-------+-----+------+
| horrible | bad | terrible | awful | wow | hate |
+----------+-----+----------+-------+-----+------+
|    0     |  0  |    0     |   0   |  0  |  0   |
|    0     |  0  |    0     |   0   |  0  |  0   |
|    0     |  0  |    0     |   0   |  0  |  0   |
|    0     |  0  |    0     |   0   |  0  |  0   |
|    1     |  0  |    0     |   0   |  0  |  0   |
|    0     |  1  |    0     |   0   |  0  |  0   |
|    0     |  0  |    0     |   0   |  0  |  0   |
|    0     |  0  |    0     |   0   |  0  |  0   |
|    0     |  0  |    0     |   0   |  0  |  0   |
|    0     |  0  |    0     |   0   |  0  |  0   |
+----------+-----+----------+-------+-----+------+
[10 rows x 16 columns]

In [159]:
diaper_champ_reviews['predicted_sentiment'] = sentiment_model.predict(diaper_champ_reviews, output_type='probability')

In [168]:
diaper_champ_reviews = diaper_champ_reviews.sort('predicted_sentiment',ascending=False)

In [169]:
diaper_champ_reviews['review','rating','predicted_sentiment'].head()

Columns:
	review	str
	rating	float
	predicted_sentiment	float

Rows: 10

Data:
+-------------------------------+--------+---------------------+
|             review            | rating | predicted_sentiment |
+-------------------------------+--------+---------------------+
| Baby Luke can turn a clean... |  5.0   |    0.999999937267   |
| I LOOOVE this diaper pail!... |  5.0   |    0.999999917406   |
| We researched all of the d... |  4.0   |    0.999999899509   |
| My baby is now 8 months an... |  5.0   |    0.999999836182   |
| This is absolutely, by far... |  5.0   |    0.999999824745   |
| Diaper Champ or Diaper Gen... |  5.0   |    0.999999759315   |
| Wow!  This is fabulous.  I... |  5.0   |    0.999999692111   |
| I originally put this item... |  5.0   |    0.999999642488   |
| Two girlfriends and two fa... |  5.0   |    0.999999604504   |
| I am one of those super-cr... |  5.0   |    0.999999486804   |
+-------------------------------+--------+---------------------+
[10 rows x 3 columns]

## Answer 4.1) Most Positive Review has Sentiment is 0.999999937267

### Predict sentiment for the most positive review
Now use the selected_words_model you learned using just the selected_words to predict the sentiment most positive review you found above. 

In [171]:
selected_words_model.predict(diaper_champ_reviews[0:1], output_type='probability')

dtype: float
Rows: 1
[0.7969408512906712]

## Answer 4.2) Probablity based on selected words only is 0.7969408512906712

In [176]:
diaper_champ_reviews[0]['review']

'Baby Luke can turn a clean diaper to a dirty diaper in 3 seconds flat. The diaper champ turns the smelly diaper into "what diaper smell" in less time than that. I hesitated and wondered what I REALLY needed for the nursery. This is one of the best purchases we made. The champ, the baby bjorn, fluerville diaper bag, and graco pack and play bassinet all vie for the best baby purchase.Great product, easy to use, economical, effective, absolutly fabulous.UpdateI knew that I loved the champ, and useing the diaper genie at a friend\'s house REALLY reinforced that!! There is no comparison, the chanp is easy and smell free, the genie was difficult to use one handed (which is absolutly vital if you have a little one on a changing pad) and there was a deffinite odor eminating from the genieplus we found that the quick tie garbage bags where the ties are integrated into the bag work really well because there isn\'t any added bulk around the sealing edge of the champ.'

In [179]:
diaper_champ_reviews.head(1)

Columns:
	name	str
	review	str
	rating	float
	sentiment	int
	word_count	dict
	awesome	int
	great	int
	fantastic	int
	amazing	int
	love	int
	horrible	int
	bad	int
	terrible	int
	awful	int
	wow	int
	hate	int
	predicted_sentiment	float

Rows: 1

Data:
+-------------------------+-------------------------------+--------+-----------+
|           name          |             review            | rating | sentiment |
+-------------------------+-------------------------------+--------+-----------+
| Baby Trend Diaper Champ | Baby Luke can turn a clean... |  5.0   |     1     |
+-------------------------+-------------------------------+--------+-----------+
+-------------------------------+---------+-------+-----------+---------+------+
|           word_count          | awesome | great | fantastic | amazing | love |
+-------------------------------+---------+-------+-----------+---------+------+
| {'all': 1, 'less': 1, "fri... |    0    |   0   |     0     |    0    |  0   |
+-------------------------------+---------+-------+-----------+---------+------+
+----------+-----+----------+-------+-----+------+---------------------+
| horrible | bad | terrible | awful | wow | hate | predicted_sentiment |
+----------+-----+----------+-------+-----+------+---------------------+
|    0     |  0  |    0     |   0   |  0  |  0   |    0.999999937267   |
+----------+-----+----------+-------+-----+------+---------------------+
[1 rows x 17 columns]

In [181]:
diaper_champ_reviews['word_count'].head(1)

dtype: dict
Rows: 1
[{'all': 1, 'less': 1, "friend's": 1, '(which': 1, 'absolutly': 2, 'to': 3, 'easy': 2, 'around': 1, 'deffinite': 1, 'luke': 1, 'champ': 1, 'turns': 1, 'bag': 1, 'quick': 1, 'found': 1, 'where': 1, "isn't": 1, 'because': 1, 'house': 1, 'are': 1, 'best': 2, 'really': 3, '"what': 1, 'what': 1, 'for': 2, 'product,': 1, 'seconds': 1, '3': 1, 'integrated': 1, 'dirty': 1, 'we': 2, 'pad)': 1, 'odor': 1, 'use': 1, 'flat.': 1, 'on': 1, 'of': 2, 'chanp': 1, 'turn': 1, 'free,': 1, 'purchase.great': 1, 'reinforced': 1, 'garbage': 1, 'vie': 1, 'into': 2, 'one': 3, 'economical,': 1, 'smelly': 1, 'ties': 1, 'nursery.': 1, 'little': 1, 'from': 1, 'there': 3, 'bjorn,': 1, 'needed': 1, 'was': 2, 'that': 2, 'smell"': 1, 'bulk': 1, 'fabulous.updatei': 1, 'hesitated': 1, 'graco': 1, 'baby': 3, 'champ,': 2, 'champ.': 1, 'than': 1, 'loved': 1, 'this': 1, 'work': 1, 'useing': 1, 'can': 1, 'pack': 1, 'and': 6, 'purchases': 1, 'bassinet': 1, 'is': 4, 'use,': 1, 'at': 1, 'have': 1, 'in': 2, 'a

None of the words appear.  There are variations on the words but not the exact words.  E.g. Loved instead of love. There are lots of speling mistakes also.